### PR TITLE
DDP-4673 add Create Study Participant API

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

New API for study admin to create a new participant, so CRCs can help patient begin enrollment.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [x] I have written automated positive tests
- [x] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/src/main/java/org/broadinstitute/ddp/DataDonationPlatform.java [hunk 1]
```diff
@@ -312,13 +301,30 @@ private static void start() {
         }
 
         post(API.REGISTRATION, new UserRegistrationRoute(interpreter), responseSerializer);
-
         post(API.TEMP_USERS, new CreateTemporaryUserRoute(), responseSerializer);
 
+        // Admin APIs
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/DataDonationPlatform.java:306]: Have a better feeling for how to layout the admin APIs, so I moved and renamed a bunch of things.

File: pepper-apis/docs/specification/src/endpoints/admin.studies.participants.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          schema:
+            allOf:
+              - $ref: '../pepper.yml#/components/schemas/Error'
+              - type: object
+                properties:
+                  code:
+                    type: string
+                    enum:
+                      - BAD_PAYLOAD
+                      - INVALID_INVITATION
```
Comment (MocanaAtBroad) [inline pepper-apis/docs/specification/src/endpoints/admin.studies.participants.yml:50]: Should we specify here unauthorized access to this API? I believe that would come from filter, right? (this is question not request).

File: pepper-apis/src/main/java/org/broadinstitute/ddp/util/RouteUtil.java [hunk 3]
```diff
@@ -114,6 +113,21 @@ public static String parseStudyGuid(String path) {
...
+     * @param path the admin URI path
+     * @return the study guid or null if the study could not be parsed
+     */
+    public static String parseAdminStudyGuid(String path) {
+        List<String> parts = SparkUtils.convertRouteToList(path);
+        String studyGuid = null;
+        if (parts.size() > ADMIN_STUDY_GUID_IDX && STUDIES_PATH_MARKER.equals(parts.get(ADMIN_STUDIES_MARKER_IDX))) {
+            studyGuid = parts.get(ADMIN_STUDY_GUID_IDX);
+        }
+        return studyGuid;
```
Comment (MocanaAtBroad) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/util/RouteUtil.java:128]: Geez! Basic path parsing based on templates and extracting variables has been in every web framework I have seen except Spark.

**Review by MocanaAtBroad — APPROVED** [review]

File: pepper-apis/docs/specification/src/endpoints/admin.studies.participants.yml (hunk 2 above)
Comment (yufengwng) [reply pepper-apis/docs/specification/src/endpoints/admin.studies.participants.yml:50]: That will come from the filter.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/util/RouteUtil.java (hunk 3 above)
Comment (yufengwng) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/util/RouteUtil.java:128]: They have it, but last time I checked they don't expose it...

**Pull request merged**